packages/react-app/src/components/giveaways/Card.tsx

```diff
@@ -77,17 +80,41 @@ const GiveawayCard = ({
       ActionButtonComponents[participationState],
       props
     );
-  }, [participationState, giveaway, onParticipationError, userInfo]);
+  }, [
+    participationState,
+    giveaway,
+    onParticipationError,
+    userInfo,
+    participants,
+  ]);
+
+  const getWinnerStr = () => {
```

Comment:
Shouldn't we just have the names of all winners, separated by commas and apply `text-overflow: ellipsis` in case there are more than what we can show without breaking the card? Also, it's better to generate a new winners array with the names of the participants in the backend. The frontend should only manipulate data if that requires user input, data that the API doesn't have access to, or if we don't have control over the API results to keep the number of calculations in the client side to a minimum

Reply:
Got it! If the `Name +2 more` format is consisted with the design requirements and it looks good, we can roll with that 👌. But let's not forget about data consistency and keeping things light on the client side 🙏.  By handling the winners array generation on the backend, we have more control over the data manipulation. It also ensures consistency across all the frontends that use the API. So, if we can do that in this case, it's gonna be smoother and more efficient.

Reply:
I took note of that improvement. I will move forward as it is since this is not a priority given the level of impact of the change, will be back on it in a separate task.